e2e/native-esm/package.json

```diff
@@ -1,5 +1,8 @@
 {
   "type": "module",
+  "devDependencies": {
+    "discord.js": "14.3.0"
```

Comment:
I can't be bothered to recreate this case, just installing the module which reproduced is easier 😅 

Essentially: ESM imports CJS, which re-exports (via CJS) modules which use `exports` to differentiate between ESM and CJS. Jest then picked out ESM when it _should_ have picked out CJS